samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+
+/* The columns (temperature ranges) for relative
+ * runtime in a chip temperature
+ */
+static const struct bt_mesh_sensor_column columns[] = {
+	{ { 0 }, { 20 } },
+	{ { 20 }, { 25 } },
+	{ { 25 }, { 30 } },
+	{ { 30 }, { 100 } },
+};
+
+static struct device *dev;
+static u32_t tot_temp_samps;
+static u32_t col_samps[ARRAY_SIZE(columns)];
+
+static s32_t prev_pres;
+
+static int chip_temp_get(struct bt_mesh_sensor *sensor,
+			 struct bt_mesh_msg_ctx *ctx, struct sensor_value *rsp)
+{
+	int err;
+
+	sensor_sample_fetch(dev);
+
+	err = sensor_channel_get(dev, SENSOR_CHAN_DIE_TEMP, rsp);
+
+	/* Should match the columns[] values */
+	if (rsp->val1 >= 0 && rsp->val1 <= 20) {
+		col_samps[0]++;
+	} else if (rsp->val1 > 20 && rsp->val1 <= 25) {
+		col_samps[1]++;
+	} else if (rsp->val1 > 25 && rsp->val1 <= 30) {
+		col_samps[2]++;
+	} else if (rsp->val1 > 30 && rsp->val1 <= 100) {
+		col_samps[3]++;
+	}
+
+	tot_temp_samps++;
+
+	return err;
+}
+
+static struct bt_mesh_sensor chip_temp = {
+	.type = &bt_mesh_sensor_present_dev_op_temp,
+	.get = chip_temp_get,
+};
+
+static int relative_runtime_in_chip_temp_get(
+	struct bt_mesh_sensor *sensor, struct bt_mesh_msg_ctx *ctx,
+	const struct bt_mesh_sensor_column *column, struct sensor_value *value)
+{
+	s32_t index = column - &columns[0];
+
+	u8_t percent_steps = (200 * col_samps[index]) / tot_temp_samps;
```

Comment:
You mentioned doing an if-check for divide by zero here. Have you not pushed it yet?

Reply:
I waited to push until I had fixed the README files. The latest code is now pushed